include/deal.II/numerics/nonlinear.h

```diff
@@ -0,0 +1,582 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/exceptions.h>
+
+#include <deal.II/sundials/kinsol.h>
+
+#include <deal.II/trilinos/nox.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * Selects a nonlinear solver by choosing between KINSOL or NOX.
+ * KINSOL and NOX are nonlinear solvers included in the SUNDIALS
+ * package and the TRILINOS package resp. IF no solver type is
+ * specified it will automaticlaly choose a solver based on what
+ * the use has installed, KINSOL having priority.
+ *
+ * By calling the @p solve function of this @p NonlinearSolverSelector, it selects the
+ * @p solve function of that @p Solver that was specified in the constructor
+ * of this class, similar to the @p SolverSelector class.
```

Comment:
This being a reference to another class, doxygen will get the mark-up right automatically:
```suggestion
 * of this class, similar to the SolverSelector class.
```